### PR TITLE
Add SAST to the code base

### LINKED ISSUE #26: Establish SAST on IaC Scan Runner project code
To keep our code clean and secure we should use SAST tools to do linting and check security.

I propose tools like pylint, flake8, mypy, pydocstyle, bandit and others.

For the development process I suggest creating a new shell script that would allow developers to run sanity, unit and integration tests before pushing anything.

I propose something similar to https://gitlab.com/gaia-x/data-infrastructure-federation-services/orc/ppr/-/blob/main/dev.sh#L20.

### PR DESCRIPTION
**Pull request by RedekG**
For SAST following tools were added:

- flake
- mypy
- pylint

This closes #26

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+# Minimum number of public methods for a class (see R0903).
+min-public-methods = 2
+
+
+[EXCEPTIONS]
+
+# Exceptions that will emit a warning when being caught. Defaults to
+# "BaseException, Exception".
+overgeneral-exceptions = BaseException,
+                         Exception
```
Comment (anzoman) [inline .pylintrc:559]: Missing newline.

File: requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+pytest==7.1.3
+requests==2.28.1
+flake8==5.0.4
+flake8-docstrings==1.6.0
+flake8-rst-docstrings==0.2.7
+mypy==0.982
+types-setuptools==65.5.0.1
+types-requests==2.28.11.2
+types-pyyaml==6.0.12
+pylint==2.15.4
```
Comment (anzoman) [inline requirements-dev.txt:10]: Please verify that these dev requirements versions are the latest ones (i.e., I know that `pytest` is at `7.4.0` now) and run the lining phase again.

File: .pylintrc [hunk 3]
```diff
@@ -0,0 +1,559 @@
...
+          consider-iterating-dictionary,
+          consider-using-dict-items,
+          too-many-instance-attributes,
+          too-many-locals,
+          too-many-branches,
+          too-many-statements,
+          too-many-return-statements,
+          no-name-in-module,
+          wrong-import-order,
+          fixme
```
Comment (anzoman) [inline .pylintrc:82]: Check if we can remove any of these. For instance, if we don't have any `FIXME` in the code we can remove `fixme` and `pylint` will still go through.

File: .flake8 [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+max-doc-length = 120
+
+exclude = .venv/
+
+# flake8-import-order
+application-import-names = src
+application-package-names =
+import-order-style = pep8
+
+ignore = D100,D101,D102,D103,D104,D105,D106,D107,D205,D301,D402,E731
```
Comment (anzoman) [inline .flake8:12]: Do we really have to ignore all these? Please check - ideally, we should not ignore much.

**Review by anzoman — CHANGES_REQUESTED** [review]
Thanks, @RedekG, looks good. I just have some comments that we need to address.

File: .pylintrc (hunk 1 above)
Comment (RedekG) [reply .pylintrc]: Fixed.

File: requirements-dev.txt (hunk 2 above)
Comment (RedekG) [reply requirements-dev.txt]: Updated the versions.

File: .pylintrc (hunk 3 above)
Comment (RedekG) [reply .pylintrc]: I cleared ones that we don't need.

File: .flake8 (hunk 4 above)
Comment (RedekG) [reply .flake8]: I cleared this too but following will remain:

D301 - If we use r"""""", function parameters will be shown in swagger UI. 
D205 - Error caused by \f  between summary line and description. 
D107 - No need to have docstring in init.
D104 - No need to have docstring in public package aka. __init__.py
D100 - No need to have docstring in public module

Comment (anzoman) [reply .flake8]: Okay, then let's keep `D301` and `D205`.

`D100`, `D104`, and `D107` can be quickly resolved by just removing these unneeded docstrings.

**Review by anzoman — APPROVED** [review]
Thanks @RedekG, looks okay, let's merge this one!

**Comment by cankarm** [comment]
Even though that all review request are being met, the Git says that still some changes need to be made. Can you check this @RedekG?

**Pull request merged**